test/index.js

```diff
@@ -108,3 +109,79 @@ test('binary encoding, using batch', function (t) {
     })
   })
 })
+
+test('default encoding retrieves a string from underlying store', function (t) {
+  t.plan(1)
+
+  var down = {
+    get: function (key, options, cb) {
+      t.is(options.asBuffer, false, '.asBuffer is false')
+    }
+  }
+
+  var db = encdown(down)
+
+  db.get('key', noop)
+})
+
+test('custom value encoding that retrieves a string from underlying store', function (t) {
+  t.plan(1)
+
+  var down = {
+    get: function (key, options, cb) {
+      t.is(options.asBuffer, false, '.asBuffer is false')
+    }
+  }
+
+  var db = encdown(down, {
+    valueEncoding: {
+      buffer: false
+    }
+  })
+
+  db.get('key', noop)
+})
+
+test('custom value encoding that retrieves a buffer from underlying store', function (t) {
+  t.plan(1)
+
+  var down = {
+    get: function (key, options, cb) {
+      t.is(options.asBuffer, true, '.asBuffer is true')
+    }
+  }
+
+  var db = encdown(down, {
+    valueEncoding: {
+      buffer: true
+    }
+  })
+
+  db.get('key', noop)
+})
+
+test('iterator options .keyAsBuffer and .valueAsBuffer defaults', function (t) {
+  t.plan(2)
+
+  var down = {
+    iterator: function (options) {
+      t.is(options.keyAsBuffer, true, '.keyAsBuffer defaults to true')
+      t.is(options.valueAsBuffer, true, '.valueAsBuffer defaults to true')
+    }
+  }
+
+  encdown(down).iterator()
+})
+
+test('iterator options .keyAsBuffer and .valueAsBuffer as strings', function (t) {
+  t.plan(2)
+
+  var down = {
+    iterator: function (options) {
+      t.is(options.keyAsBuffer, false, '.keyAsBuffer should be false')
+      t.is(options.valueAsBuffer, false, '.valueAsBuffer should be false')
+    }
+  }
+
+  encdown(down).iterator({ keyAsBuffer: false, valueAsBuffer: false })
```

Comment:
Can you write this in the same way as the `get()` test? Becauxe the `keyAsBuffer` and `valueAsBuffer` options should also be based on `encoding.buffer`. 

We used to have this code in `levelup`:

```js
options = this._codec.encodeLtgt(options)		
options.keyAsBuffer = this._codec.keyAsBuffer(options)		
options.valueAsBuffer = this._codec.valueAsBuffer(options)
```

Reply:
Sure!

Reply:
If you don't mind, I'll add an _additional_ test and keep this one as it is.

Reply:
Got me thinking. Big picture, if you use `encoding-down`, then `*asBuffer` is an internal property determined by encoding. But if you don't, then `asBuffer` is a public option, like in [this `levelup` test](https://github.com/Level/levelup/blob/a4cd129b3eb3ceb15d798137fefd531651ffa070/test/no-encoding-test.js#L33). Perhaps that needs clarification in docs (eventually, as most people *will* use `encoding-down`), but where?

Reply:
@vweevers Good point. The idea struck my mind as well. Make an issue for it for now so we can move out the discussion from this PR?